src/utils/log.c

```diff
@@ -32,15 +33,16 @@ void log_hide(char mode, const char *fn, const char *fmt, ...) {
 void log_print(char mode, const char *fn, const char *fmt, ...) {
     if(handle == 0)
         return;
+    int len;
     if(fn != NULL) {
-        fprintf(handle, "[%7u][%c] %s(): ", _log_tick, mode, fn);
+        len = snprintf(log_buf, sizeof(log_buf), "[%7u][%c] %s(): ", _log_tick, mode, fn);
     } else {
-        fprintf(handle, "[%7u][%c] ", _log_tick, mode);
+        len = snprintf(log_buf, sizeof(log_buf), "[%7u][%c] ", _log_tick, mode);
     }
     va_list args;
     va_start(args, fmt);
-    vfprintf(handle, fmt, args);
+    vsnprintf(&log_buf[len], sizeof(log_buf) - len, fmt, args);
     va_end(args);
-    fprintf(handle, "\n");
+    fprintf(handle, "%s\n", log_buf);
```

Comment:
Why not `fputs`?

Reply:
Well, the previous code was using this stream handle thing .. dunno

Reply:
In theory, the log could be going to a file that is not stdout.

Reply:
Oh right, fputs does have `FILE*` argument .. not sure it's worth changing ..

Reply:
Okay, so how to add the ending newline nicely if we are using fputs?

Reply:
`fputc`

Reply:
Nah, would make for 2 writes.

Reply:
That should not matter, `FILE *` functions are buffered.